tsdb/exemplar.go

```diff
@@ -361,6 +357,10 @@ func (ce *CircularExemplarStorage) AddExemplar(l labels.Labels, e exemplar.Exemp
 	ce.lock.Lock()
 	defer ce.lock.Unlock()
 
+	if len(ce.exemplars) == 0 {
```

Comment:
maybe it's better to move the lock to the beginning instead, to avoid the `buf` allocation when not needed.

Reply:
Yeah, I was honestly not sure whether it would be preferable to allocate the slice before taking the lock.

Reply:
Moved it, WDYT?